test/sslapitest.c

```diff
@@ -1863,36 +1868,141 @@ static int ocsp_server_cb(SSL *s, void *arg)
         return SSL_TLSEXT_ERR_ALERT_FATAL;
     }
 
-    if (!TEST_ptr(copy = OPENSSL_memdup(orespder, sizeof(orespder))))
+    dummy_ocsp_resp = OCSP_response_create(OCSP_RESPONSE_STATUS_SUCCESSFUL, NULL);
+
+    if (!TEST_ptr(dummy_ocsp_resp)) {
         return SSL_TLSEXT_ERR_ALERT_FATAL;
+    }
 
-    if (!TEST_true(SSL_set_tlsext_status_ocsp_resp(s, copy,
-                                                   sizeof(orespder)))) {
-        OPENSSL_free(copy);
+    resplen = i2d_OCSP_RESPONSE(dummy_ocsp_resp, &ocspresp);
+
+    OCSP_RESPONSE_free(dummy_ocsp_resp);
+
+    if (!TEST_true(SSL_set_tlsext_status_ocsp_resp(s, ocspresp, resplen))) {
+        OPENSSL_free(ocspresp);
         return SSL_TLSEXT_ERR_ALERT_FATAL;
     }
+
+    OPENSSL_free(ocspresp);
+
     ocsp_server_called = 1;
     return SSL_TLSEXT_ERR_OK;
 }
 
-static int ocsp_client_cb(SSL *s, void *arg)
+static int ocsp_client_cb_single(SSL *s, void *arg)
 {
     int *argi = (int *)arg;
-    const unsigned char *respderin;
-    size_t len;
+    const unsigned char *resp, *p;
+    OCSP_RESPONSE *rsp;
+    int len, i;
 
     if (*argi != 1 && *argi != 2)
         return 0;
 
-    len = SSL_get_tlsext_status_ocsp_resp(s, &respderin);
-    if (!TEST_mem_eq(orespder, len, respderin, len))
+    len = SSL_get_tlsext_status_ocsp_resp(s, &resp);
+
+    p = resp;
+    rsp = d2i_OCSP_RESPONSE(NULL, &p, len);
+
+    i = OCSP_response_status(rsp);
+
+    OCSP_RESPONSE_free(rsp);
+
+    if (i != OCSP_RESPONSE_STATUS_SUCCESSFUL)
         return 0;
 
     ocsp_client_called = 1;
     return 1;
 }
 
-static int test_tlsext_status_type(void)
+static int ocsp_server_cb_multi(SSL *s, void *arg)
+{
+    STACK_OF(OCSP_RESPONSE) *sk_resp = NULL;
+    int *argi = (int *)arg;
+    STACK_OF(OCSP_RESPID) *ids = NULL;
+    OCSP_RESPID *id = NULL;
+
+    if (*argi == 2) {
+        /* In this test we are expecting exactly 1 OCSP_RESPID */
+        SSL_get_tlsext_status_ids(s, &ids);
+        if (ids == NULL || sk_OCSP_RESPID_num(ids) != 1) {
+            return SSL_TLSEXT_ERR_ALERT_FATAL;
+        }
+
+        id = sk_OCSP_RESPID_value(ids, 0);
+        if (id == NULL || !OCSP_RESPID_match_ex(id, ocspcert, libctx, NULL)) {
+            return SSL_TLSEXT_ERR_ALERT_FATAL;
+        }
+    } else if (*argi != 1) {
+        return SSL_TLSEXT_ERR_ALERT_FATAL;
+    }
+
+    dummy_ocsp_resp = OCSP_response_create(OCSP_RESPONSE_STATUS_SUCCESSFUL, NULL);
+
+    if (!TEST_ptr(dummy_ocsp_resp)) {
+        return SSL_TLSEXT_ERR_ALERT_FATAL;
+    }
+
+    sk_resp = sk_OCSP_RESPONSE_new_null();
+    sk_OCSP_RESPONSE_push(sk_resp, dummy_ocsp_resp);
+
+    if (!TEST_true(SSL_set_tlsext_status_ocsp_resp_ex(s, sk_resp))) {
+        return SSL_TLSEXT_ERR_ALERT_FATAL;
+    }
```

Comment:
Here several parts violate the coding style: do not use braces around single statements.
Again, better use `check-format.pl` to avoid such glitches.

Reply:
I checked with `check-format.pl`, but it gives me no hint about the braces.

Reply:
Ah, right, in this case the `check-format.pl` checks were incomplete as long as the `-1` option is not given.
Fix for that is in https://github.com/openssl/openssl/pull/24805.